fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveMetastore.java

```diff
@@ -231,6 +231,10 @@ public HivePartitionStats getTableStatistics(String dbName, String tblName) {
                 .map(FieldSchema::getName)
                 .collect(toImmutableList());
         List<ColumnStatisticsObj> statisticsObjs = client.getTableColumnStats(dbName, tblName, dataColumns);
+        if (statisticsObjs.isEmpty()) {
```

Comment:
Is the partition level statistics also in the table properties?

Reply:
1. For a partitioned table:

`create table testsparkstats (id int, name string) partitioned by(dt string) tored as orc;`

2. insert two partition dt=2024 and dt=2025

`insert into testsparkstats values(1243,'ttetw','2024'),(8945,'erw','2024'),(456,'df','2025'),(568,'tom','2025');`

3. analyze column statistics against the partitioned table

**For `column `level statistics, spark will store all the column statistics in the table properties whether the table is partitioned or not, that is to say, the two statements have same effect (spark will treat partitioned table as unpartitioned table when doing `column `statistics)**
1) analyze the partition column statistics dt=2024, and this will get all table's statistics and stored in the table properties
`analyze table testsparkstats partition (dt='2024') compute statistics for all columns;`
2) analyze partitoned table's level statistics 
`analyze table testsparkstats compute statistics for all columns;`

mysql> select * from TABLE_PARAMS where TBL_ID=1082;
+--------+--------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| TBL_ID | PARAM_KEY                                        | PARAM_VALUE                                                                                                                                                                                                         |
+--------+--------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
|   1082 | spark.sql.create.version                         | 3.3.1                                                                                                                                                                                                               |
|   1082 | spark.sql.sources.schema                         | {"type":"struct","fields":[{"name":"id","type":"integer","nullable":true,"metadata":{}},{"name":"name","type":"string","nullable":true,"metadata":{}},{"name":"dt","type":"string","nullable":true,"metadata":{}}]} |
|   1082 | spark.sql.sources.schema.numPartCols             | 1                                                                                                                                                                                                                   |
|   1082 | spark.sql.sources.schema.partCol.0               | dt                                                                                                                                                                                                                  |
|   1082 | spark.sql.statistics.colStats.dt.avgLen          | 4                                                                                                                                                                                                                   |
|   1082 | spark.sql.statistics.colStats.dt.distinctCount   | 2                                                                                                                                                                                                                   |
|   1082 | spark.sql.statistics.colStats.dt.maxLen          | 4                                                                                                                                                                                                                   |
|   1082 | spark.sql.statistics.colStats.dt.nullCount       | 0                                                                                                                                                                                                                   |
|   1082 | spark.sql.statistics.colStats.dt.version         | 2                                                                                                                                                                                                                   |
|   1082 | spark.sql.statistics.colStats.id.avgLen          | 4                                                                                                                                                                                                                   |
|   1082 | spark.sql.statistics.colStats.id.distinctCount   | 4                                                                                                                                                                                                                   |
|   1082 | spark.sql.statistics.colStats.id.max             | 8945                                                                                                                                                                                                                |
|   1082 | spark.sql.statistics.colStats.id.maxLen          | 4                                                                                                                                                                                                                   |
|   1082 | spark.sql.statistics.colStats.id.min             | 456                                                                                                                                                                                                                 |
|   1082 | spark.sql.statistics.colStats.id.nullCount       | 0                                                                                                                                                                                                                   |
|   1082 | spark.sql.statistics.colStats.id.version         | 2                                                                                                                                                                                                                   |
|   1082 | spark.sql.statistics.colStats.name.avgLen        | 4                                                                                                                                                                                                                   |
|   1082 | spark.sql.statistics.colStats.name.distinctCount | 4                                                                                                                                                                                                                   |
|   1082 | spark.sql.statistics.colStats.name.maxLen        | 5                                                                                                                                                                                                                   |
|   1082 | spark.sql.statistics.colStats.name.nullCount     | 0                                                                                                                                                                                                                   |
|   1082 | spark.sql.statistics.colStats.name.version       | 2                                                                                                                                                                                                                   |
|   1082 | spark.sql.statistics.numRows                     | 4                                                                                                                                                                                                                   |
|   1082 | spark.sql.statistics.totalSize                   | 762                                                                                                                                                                                                                 |
|   1082 | transient_lastDdlTime                            | 1706411426                                                                                                                                                                                                          |
+--------+--------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
24 rows in set (0.00 sec)
`

Reply:
> Is the partition level statistics also in the table properties?

@stephen-shelby Therefore, to be more specific, spark does not have partition's level `column `statistics information. It only have table's level statistics information.

Reply:
If the current table statistics are from the past month, and I am currently analyzing today's partition, are the table properties in the table aggregated, such as ndv?

Reply:
Yes, almost. If you analyze today's partition, you will get the table's level column statistics(**aggregate all the partitions columns statistics**). As i said above, `spark does not have single partition's level column statistics information. It only have table's level statistics information.`, **analyze partiton column statistics** is equal **analyze table column statistics**.

Reply:
starrocks has supported `analyle table` in 3.2. We recommend using our analyze. :smile:

Reply:
It's good that SR has own analyze feat. But this pr is just to reuse the existing column statics if users don't want or forget to analyze hive table. And then we can try best to optimize the query using the existing spark column statistics in case of no analyzing.😀

Reply:
ok. got it, you need to adapt it in the `HiveMetastore#getPartitionStatistics` when it's a partitioned table.

Reply:
Since Spark can't generate partition level column stats, we can only reuse basic stats generated by spark (numRows, totalSize).
Basic stats: `HiveMetastoreApiConverter::toHiveCommonStats` (https://github.com/StarRocks/starrocks/pull/40024/files#diff-96bde93fa567de13289e23b85d008257c0af92b5578d769568f266738622870dR578-R584)

And `HiveMetastore#getPartitionStatistics` will call this method `HiveMetastoreApiConverter::toHiveCommonStats`:
https://github.com/StarRocks/starrocks/blob/654d6b73b5080cbc8055f27c6085b428ea362965/fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveMetastore.java#L290

Reply:
partitioned table don't use getTableStatistics. ref : https://github.com/StarRocks/starrocks/blob/main/fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveStatisticsProvider.java#L80